### PR TITLE
Polymer Elements Tutorial

### PR DESCRIPTION
**Pull request by manolo**
A tutorial for creating modern apps using external elements. In this case we use polymer elements wrapped with JsInterop interfaces trough the gwt-polymer-elements library.
The user learns how to create maven projects from scratch, run superdevmode, import external libraries, and design interfaces with UIBinder using either Widgets or Elements.

[Review on Reviewable](https://reviewable.io/reviews/gwtproject/gwt-site/94)

### CODE REVIEW

**Comment by christiangoudreau** [comment]
Great example. I don't want to be a party cracker, but I don't think we should include external libraries unless owned by the Steering committee.

**Comment by jDramaix** [comment]
Good point, you guys should at least discuss that in the next steering committee meeting

**Comment by jnehlmeier** [comment]
Yeah I thought the same when reading on gwt-contrib. 

Somehow I feel like gwtproject.org should teach you how to create something like a PaperButtonElement/Widget on your own, instead of teaching you how to use a 3rd party library to build a todo app. The 3rd party library kind of feels like an "official" recommendation now.

Of course having an additional tutorial on how to create a GWT app, in this case a Todo list, is great. But the tutorial as-is would be more suitable as documentation of the gwt-polymer-elements project.

**Comment by manolo** [comment]
Well, if i understood correctly the gwt-meetup conversations, the tendency in GWT is to extract from GWT core almost everything but the compiler, so things like RPC, RF,  etc would become external libraries, I don't see any reason for not keeping that documentation in gwtproject.org.

**Comment by manolo** [comment]
> owned by the Steering committee.
Does the SC want to own other libraries but GWT?
Also is impossible that SC owns other things like Paper, Iron ...

**Comment by jDramaix** [comment]
> Well, if i understood correctly the gwt-meetup conversations, the tendency in GWT is to extract from GWT core almost everything but the compiler, so things like RPC, RF, etc would become external libraries, I don't see any reason for not keeping that documentation in gwtproject.org.

external libs officially supported by the gwt-steering committee.

**Comment by jnehlmeier** [comment]
I really understand you @manolo . 

But reality is that there seems to be nobody in charge of the actual content / structure of gwtproject.org which in turn means there is little to no thought process going on to create a good structure with matching up-to-date content that gives new GWT users a great experience learning GWT.

For example from my understanding the current tutorial section is meant to be a "Lets get started - Write your first GWT app in 15 minutes" tutorial. Now with your additions we would have two applications to get started with and your tutorial also pulls in polymer as a second thing to be aware of (not necessarily a bad thing). Also the placement of the menu item is a bit unlucky because all items under "Tutorial" reference the Stockwatcher app. Now there is a last menu item which contains an entire new tutorial which IMHO is bad structuring.

Personally I feel like we should not throw all kinds of tutorials under the "Tutorial" menu item. These stuff will get outdated at some point as no one will probably keep them up-to-date. GWT documentation must be very focused to reduce the amount of outdated content.

So at the end I would vote for modernizing the stock watcher walkthrough or replacing it with an entire new one like your TodoList tutorial. If it should contain the polymer library or not is discussable but one has to be aware that using a polymer lib in a "getting started" tutorial feels like a recommendation.

**Comment by jojule** [comment]
I agree with @jnehlmeier that gwtproject.org would really need someone who would have authority over the site, would make decisions and would commit in investing time for renewing the contents. Because we are not there, we should not meanwhile try to kill the little contributions we get.

This tutorial shows how to building something modern and cool with GWT using 100% open source (Apache licensed) stack. While it is a 3rd party library (by Google and with thin wrapper by Vaadin), wouldn't most parts of the GWT be outside of the core soon? And wouldn't large part of the site today be something that should not be added to the site if it would be contributed today?

**Comment by christiangoudreau** [comment]
I want to apologize for the bit of passion my comment generated! There's a lot of valid points from everyone involved. I do value contributions that makes GWT better and will continue to do so.

File: src/main/markdown/doc/latest/polymer-tutorial/introduction.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+## Introduction
+
+In this tutorial, you’ll learn how to write a **TodoList** application using [Polymer][0] [Web Components][1]. Web Components define a collection of standards allowing us to bundle markup and styles into custom HTML elements. Polymer is a JS library to make easier deal with Web Components.
+
+In this case, we want to fulfil the [Material Design][2] specification. Therefore, we’ll be using the Vaadin [gwt-polymer-elements][3] library. This library, generated by [gwt-api-generator][4], is a JsInterop wrapper for the Polymer [Iron and Paper][5] elements.
+
+Before we start, you might want to try out the [TodoList][6] application to see what we are about to build.
+
+## **Disclaimer**:
```
Comment (jDramaix) [inline src/main/markdown/doc/latest/polymer-tutorial/introduction.md]: I would put this diclaimer on top of each pages of the tutorial.

Comment (manolo) [inline src/main/markdown/doc/latest/polymer-tutorial/introduction.md]: IMO, it is very redundant to have the same section in 6 pages.
I think the disclaimer is very clear in the introduction page, and it is clear the position of the gwtproject related with external libraries. Repeating the disclaimer 6 times does not add more info but noise.

Comment (jDramaix) [inline src/main/markdown/doc/latest/polymer-tutorial/introduction.md]: People can land on a page without passing by the introduction page

Comment (jDramaix) [inline src/main/markdown/doc/latest/polymer-tutorial/introduction.md]: you can put it a the end of the page if you found it's too noisy

Comment (manolo) [inline src/main/markdown/doc/latest/polymer-tutorial/introduction.md]: The disclaimer should be something brief, standard and well designed then (some special css), and we must put it in many sites apart from this tutorial pages, like in those pages speaking about ant vs maven, php vs ruby, etc.

**Comment by manolo** [comment]
@jDramaix finally added a disclaimer link at the end of each page, it's not so noisy and we don't have to copy all the disclaimer in each file.
You can see a clone of the documentation with these chantes live at http://localhost:3000/doc/latest/RefCommandLineTools.html

File: src/main/markdown/doc/latest/polymer-tutorial/introduction.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Building a modern GWT app using Polymer Elements:
+
+## **Disclaimer** <a id="pol-disclaimer"></a>:
+_This tutorial is based on the usage of three 3rd party **open-source** libraries out of the **gwtproject** organization_:
```
Comment (jDramaix) [inline src/main/markdown/doc/latest/polymer-tutorial/introduction.md]: third party

File: src/main/markdown/doc/latest/polymer-tutorial/introduction.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+* _[**Polymer**][1]: a javascript library for building web applications out of web components and owned by **Google Inc**._
+* _[**Polymer Elements**][5]: a collection of widgets built in Polymer owned by **Google Inc**._
+* _[**GWT-Polymer-Elements**][3]: a Java wrapper which allows to use Polymer Elements in GWT projects, it's owned by **Vaadin Ltd**._
+
+The main goal of the tutorial is to teach our audience the full process of creating real GWT projects interacting with real JS stuff. But **gwtproject** does not want to recomend the usage of any of those components.
```
Comment (jDramaix) [inline src/main/markdown/doc/latest/polymer-tutorial/introduction.md]: recommend

**Comment by jDramaix** [comment]
nits then LGTM!
awesome work :)

**Pull request merged**